solr/test-framework/src/java/org/apache/solr/cloud/AbstractFullDistribZkTestBase.java

```diff
@@ -2283,6 +2292,26 @@ protected CloudSolrClient getCommonCloudSolrClient() {
     return commonCloudSolrClient;
   }
 
+  protected CloudSolrClient getSolrClientForCollection(String collectionName) {
+    synchronized (this) {
+      if (!solrClientForCollectionCache.containsKey(collectionName)) {
```

Comment:
No, use `computeIfAbsent` instead of synchronization, since you chose to use ConcurrentHashMap (which makes sense).  It's faster and safer.  This may explain why Lift/Infer complained because you are using synchronized here but not in destroyServers when you called clear() on the map.

Reply:
thanks for explaining, and I just pushed up the change.

Reply:
Did you push up a change @epugh? Still see synchronized